document/document.go

```diff
@@ -75,6 +75,34 @@ func Fields(d Document) ([]string, error) {
 	return fields, nil
 }
 
+func IterateInOrder(d Document, fn func(string, Value) error) error {
```

Comment:
I'm not sure this is necessary, as 60% of this function is already achieved by `document.FieldBuffer` and the `document.Fields` function:

```go
fields, _ := document.Fields(d)
for _, field := range fields {
  v, _ := d.GetByField(field)
}
``` 

Also, `document.Fields` automatically detect if the document is a `FieldBuffer` and take the short path by calling `FieldBuffer.Fields()`